core/iwasm/aot/aot_runtime.c

```diff
@@ -3366,39 +3366,45 @@ aot_table_fill(AOTModuleInstance *module_inst, uint32 tbl_idx, uint32 length,
 }
 
 uint32
-aot_table_grow(AOTModuleInstance *module_inst, uint32 tbl_idx,
-               uint32 inc_entries, table_elem_type_t init_val)
+aot_table_grow(AOTModuleInstance *module_inst, uint32 tbl_idx, uint32 inc_size,
+               table_elem_type_t init_val)
 {
-    uint32 entry_count, i, orig_tbl_sz;
     AOTTableInstance *tbl_inst;
+    uint32 i, orig_size, total_size;
 
     tbl_inst = module_inst->tables[tbl_idx];
     if (!tbl_inst) {
         return (uint32)-1;
     }
 
-    orig_tbl_sz = tbl_inst->cur_size;
+    orig_size = tbl_inst->cur_size;
 
-    if (!inc_entries) {
-        return orig_tbl_sz;
+    if (!inc_size) {
+        return orig_size;
     }
 
-    if (tbl_inst->cur_size > UINT32_MAX - inc_entries) {
+    if (tbl_inst->cur_size > UINT32_MAX - inc_size) {
+        LOG_ERROR("table grow (%d -> %d) failed because of integer overflow",
+                  tbl_inst->cur_size, inc_size);
+        aot_set_exception_with_id(module_inst, EXCE_INTEGER_OVERFLOW);
```

Comment:
Like `memory.grow`, no need to throw the exception for `table.grow`, just return -1 when failed.

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_interp_classic.c#L5808-L5813

Reply:
how about a LOG_WARNING, or it really needs a long time to figure out

Reply:
Yes, had better just use LOG_WARNING and don't throw exception.